nifi-nar-bundles/nifi-alluxio-bundle/nifi-alluxio-processors/src/main/java/org/apache/nifi/processors/alluxio/GetAlluxio.java

```diff
@@ -0,0 +1,184 @@
+            .autoTerminateDefault(true)
+            .build();
+
+    private final static Set<Relationship> relationships;
+
+
+    /*
+     * Will ensure that the list of property descriptors is build only once.
+     * Will also create a Set of relationships
+     */
+    static {
+        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
+        _propertyDescriptors.addAll(descriptors);
+        _propertyDescriptors.add(READ_TYPE);
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        _relationships.add(REL_SUCCESS_REQ);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        createFileSystem(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile request = null;
+        if (context.hasIncomingConnection()) {
+            request = session.get();
+
+            // If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+            // However, if we have no FlowFile and we have connections coming from other Processors, then
+            // we know that we should run only if we have a FlowFile.
+            if (request == null && context.hasNonLoopConnection()) {
+                return;
+            }
+        }
+
+        final StopWatch stopWatch = new StopWatch(true);
+        final String uri = context.getProperty(URI).evaluateAttributeExpressions(request).getValue();
+        final AlluxioURI path = new AlluxioURI(uri);
+        final OpenFileOptions options = OpenFileOptions.defaults().setReadType(ReadType.valueOf(context.getProperty(READ_TYPE).getValue()));
+
+        FileInStream in = null;
+        FlowFile flowFile = null;
+
+        if(request == null) {
+            flowFile = session.create(request);
+        } else {
+            flowFile = session.create();
+        }
+
+        try {
+            final URIStatus status = fileSystem.get().getStatus(path);
+            flowFile = updateFlowFile(status, flowFile, session);
+
+            in = fileSystem.get().openFile(path, options);
+            final FileInStream toCopy = in;
+
+            flowFile = session.write(flowFile, new OutputStreamCallback() {
+                @Override
+                public void process(final OutputStream out) throws IOException {
+                    IOUtils.copy(toCopy, out);
```

Comment:
I'll have to scope this out again.  After the fact I realized that it may have just been the bulletin bug that was on master that was causing bulletins to work.  When I next get the opportunity, I'll see if that's still the case.

For some additional context, in the logs, which I now realize I failed to capture, there was mention of the Alluxio logger.  I was curious if that was somehow interfering with NiFi's logger and preventing bulletins.  I'll let you know if I can recreate.